apis/infrastructure/v1beta1/byohost_types.go

```diff
@@ -32,6 +32,26 @@ type ByoHostSpec struct {
 	BootstrapSecret *corev1.ObjectReference `json:"bootstrapSecret,omitempty"`
 }
 
+// HostStatus is a set of details about the host platform.
+type HostStatus struct {
+	// MachineID reported by the host. For unique machine identification
+	// in the cluster this field is preferred. Learn more from man(5)
+	// machine-id: http://man7.org/linux/man-pages/man5/machine-id.5.html
+	HostID string `json:"hostid,omitempty"`
```

Comment:
Why do we need HostID and KernelVersion at this point?

Reply:
We discussed these fields in the corresponding issue (#479). 
We need some unique identifier like a machine-id (for Linux only) to identify the particular host in the workload pool. Similarly, if we already have an OS in place, having info about its kernel is useful in many cases when we need to know the compatibility of certain dependencies.
Makes sense?

Reply:
> We need some unique identifier like a machine-id (for Linux only) to identify the particular host in the workload pool

How is this information consumed by the provider or the user?

> having info about its kernel is useful in many cases when we need to know the compatibility of certain dependencies

What dependencies? I'd hold off adding this field until we have some need / requirement from the user.

Reply:
> How is this information consumed by the provider or the user?

Hmm. I was thinking in reference to targeting a specific host for something (like removing a specific host [#288]) but now I think we can just use the name of the particular host (as assigned after registration) for such things so we can remove this.

> What dependencies?

Like some driver support or some new improvements that are made in the latest versions that were not available in the previous kernel versions.

Reply:
Yup, I agree with @anusha94 and think we can introduce these 2 fields at a later point in time as and when required and in the PR let's just do the minimal
( I know adding these fields will provide more info about the host, but we are not sure how are we going to use this).